sphinx/ext/doctest.py

```diff
@@ -366,6 +367,39 @@ def write(self, build_docnames, updated_docnames, method='update'):
             doctree = self.env.get_doctree(docname)
             self.test_doc(docname, doctree)
 
+    def get_filename_for_node(self, node, docname):
+        # type: (nodes.Node, unicode) -> str
+        """Try to get the file which actually contains the doctest, not the
+        filename of the document it's included in."""
+        try:
+            filename = (
+                path.relpath(node.source, self.env.srcdir)
+                .replace(path.sep, '/')
+                .rsplit(':docstring of ', maxsplit=1)[0]
+            )
+        except Exception:
+            filename = self.env.doc2path(docname, base=None)
+        if PY2:
+            return filename.encode(fs_encoding)
```

Comment:
Do you know why is this needed? I think this is not needed, and this should return unicode path string.